src/averageAges.js

```diff
@@ -55,10 +68,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const allChildrenOrSons = onlyWithSon === undefined
+    ? people
+    : people.filter((person) => person.sex === 'm');
+  const childrenWithMother = allChildrenOrSons.filter((child) =>
+    people.some((person) => person.name === child.mother));
+
+  const ageDifferences = childrenWithMother.map((child) => {
+    const childMother = people.find((person) => person.name === child.mother);
+
+    return child.born - childMother.born;
+  });
```

Comment:
Move this logic to the middle of the `reduce` so you don't have to go through the array twice when you can do it once.